content/200-concepts/100-components/01-prisma-schema/10-postgresql-extensions.mdx

```diff
@@ -0,0 +1,107 @@
+
+Representing PostgreSQL extensions in your Prisma schema is currently a preview feature. To enable the `postgresExtensions` preview feature, you will need to add the `postgresExtensions` feature flag to the `previewFeatures` field of the `generator` block in your Prisma schema file:
+
+```prisma file=schema.prisma highlight=3;add
+generator client {
+  provider        = "prisma-client-js"
+  previewFeatures = ["postgresExtensions"]
+}
+
+datasource db {
+  provider = "postgresql"
+  url      = env("DATABASE_URL")
+}
+```
+
+## How to represent PostgreSQL extensions in your Prisma schema
+
+To represent PostgreSQL extensions in your Prisma schema, add the `extensions` field to the `datasource` block of your `schema.prisma` file with an array of the extensions that you require. For example, the following schema lists the `hstore`, `pg_tgrm` and `postgis` extensions:
+
+```prisma file=schema.prisma
+datasource db {
+  provider = "postgresql"
+  url      = env("TEST_DATABASE_URL")
+  extensions = [hstore(schema: "myHstoreSchema"), pg_tgrm, postgis(version: "2.1")]
+}
+```
+
+Each extension name in the Prisma schema can take the following optional arguments:
+
+- `schema`: the name of the schema in which to install the extension's objects. If this argument is not specified, the current default object creation schema is used.
+- `version`: the version of the extension to install. If this argument is not specified, the value given in the extension's control file is used.
+- `map`: the database name of the extension. If this argument is not specified, the name of the extension in the Prisma schema must match the database name.
+
+In the example above, the `hstore` extension uses the `myHstoreSchema` schema, and the `postgis` extension is installed with version 2.1.
+
+The `map` argument is useful when the PostgreSQL extension you want to install has a name that is not a valid identifier in the Prisma schema. For example, the `uuid-ossp` PostgreSQL extension name is an invalid identifier because it contains a hyphen. In the following example, the extension is mapped to the valid name `uuidOssp` in the Prisma schema:
+
+```prisma file=schema.prisma
+datasource db {
+  provider = "postgresql"
+  url      = env("TEST_DATABASE_URL")
+  extensions = [uuidOssp(map: "uuid-ossp")]
+}
+```
+
+## How to introspect PostgreSQL extensions
+
+To [introspect](/concepts/components/introspection) PostgreSQL extensions currently installed in your database and add relevant extensions to your Prisma schema, run `npx prisma db pull`.
+
+Many PostgreSQL extensions are not relevant to the Prisma schema. For example, some extensions are intended for database administration tasks that do not change the schema. If all these extensions were included, the list of extensions would be very long. To avoid this, Prisma maintains an allowlist of known relevant extensions. The current allowlist is the following:
+
+- `citext`
```

Comment:
Is this still the correct list? Will there be somewhere I can link out to for the list, or will we have to duplicate it in docs?

Reply:
This will be hard coded in the code, but of course docs should present that in a more readable form with links to the extension pages, some context and so on.

Reply:
I've added some context for each one. 

Still want to confirm with @pimeys that this is the correct list.

Reply:
Have confirmed that this is the correct list